### PR TITLE
#75 Make setupDependencies private

### PR DESCRIPTION
**Pull request by jperedadnr**

### CODE REVIEW

File: src/main/java/com/gluonhq/substrate/util/FileDeps.java [hunk 1]
```diff
@@ -81,22 +82,53 @@
...
+     * Return the path to the JavaFX SDK for this configuration.
+     * The path is cached on the provided configuration.
+     * If it is not there yet, all dependencies are retrieved.
+     * @param configuration the project configuration
+     * @param path the initial path
+     * @param errorMessage a message that will be displayed in case of error
+     * @return the location of the JavaFX SDK for the arch-os for this configuration
+     * @throws IOException in case anything goes wrong.
+     */
+    private static Path resolvePath(ProjectConfiguration configuration, Path path, String errorMessage) throws IOException {
```
Comment (johanvos) [inline src/main/java/com/gluonhq/substrate/util/FileDeps.java]: add JavaFX in the name of the method. It is clear from the javadoc, but not from the method

Comment (jperedadnr) [reply src/main/java/com/gluonhq/substrate/util/FileDeps.java]: I've updated its javadoc, this is a method not JavaFX specific

File: src/main/java/com/gluonhq/substrate/util/FileDeps.java [hunk 2]
```diff
@@ -81,22 +82,51 @@
...
+     * @throws IOException in case anything goes wrong.
+     */
+    public static Path getJavaFXSDKLibsPath(ProjectConfiguration configuration) throws IOException {
+        return resolvePath(configuration,
+                configuration.getJavafxStaticLibsPath(),
+                "Fatal error, could not install JavaFX SDK ");
+    }
 
+    /**
+     * Fpr a given path, it verifies that the path exists, or else it tries
```
Comment (johanvos) [inline src/main/java/com/gluonhq/substrate/util/FileDeps.java]: Fpr (minor typo)

File: src/main/java/com/gluonhq/substrate/SubstrateDispatcher.java [hunk 3]
```diff
@@ -84,10 +84,6 @@ public static void main(String[] args) throws Exception {
         if (staticLibs != null) {
```
Comment (eugener) [inline src/main/java/com/gluonhq/substrate/SubstrateDispatcher.java]: In such cases Optional is cleaner I think:
`Optional.of(staticLibs).ifPresent(config::setStaticLibs)`

File: src/main/java/com/gluonhq/substrate/target/IosTargetConfiguration.java [hunk 4]
```diff
@@ -208,25 +208,26 @@ String getAppPath(String appName) {
...
         // we are using JavaFX
-        String javafxSDK = FileDeps.getJavaFXSDK(projectConfiguration).resolve("lib").toString();
+        String javafxSDKLibs = FileDeps.getJavaFXSDKLibsPath(projectConfiguration).toString();
 
-        StringBuffer answer = new StringBuffer();
-        Stream.of(classPath.split(File.pathSeparator)).forEach(s ->{
-            if (s.indexOf("javafx") < 0 ){
-                answer.append(s).append(File.pathSeparator);
+        StringBuilder answer = new StringBuilder();
+        Stream.of(classPath.split(File.pathSeparator)).forEach(s -> {
```
Comment (eugener) [inline src/main/java/com/gluonhq/substrate/target/IosTargetConfiguration.java]: If you're using stream, then IMO it is better to use functional flow, i.e. map each element to appropriate string and then collect to string using path separator.

File: src/main/java/com/gluonhq/substrate/target/IosTargetConfiguration.java [hunk 5]
```diff
@@ -208,25 +208,26 @@ String getAppPath(String appName) {
      */
     @Override
     String processClassPath(String classPath) throws IOException {
-        if (!this.projectConfiguration.isUseJavaFX()) {
+        if (!projectConfiguration.isUseJavaFX()) {
             return classPath;
         }
         // we are using JavaFX
-        String javafxSDK = FileDeps.getJavaFXSDK(projectConfiguration).resolve("lib").toString();
+        String javafxSDKLibs = FileDeps.getJavaFXSDKLibsPath(projectConfiguration).toString();
```
Comment (eugener) [inline src/main/java/com/gluonhq/substrate/target/IosTargetConfiguration.java]: `javafxSDKLibs` can be a `Path`, which makes it easy to build subpaths out of (using resolve) later

File: src/main/java/com/gluonhq/substrate/util/FileDeps.java [hunk 6]
```diff
@@ -81,22 +82,51 @@
...
+     * @throws IOException in case anything goes wrong.
+     */
+    private static Path resolvePath(ProjectConfiguration configuration, Path path, String errorMessage) throws IOException {
+        if (Files.exists(Objects.requireNonNull(path))) {
+            return path;
+        }
+        if (!setupDependencies(Objects.requireNonNull(configuration))) {
+            throw new IOException("Error setting up dependencies");
+        }
+        if (!Files.exists(path)) {
```
Comment (eugener) [inline src/main/java/com/gluonhq/substrate/util/FileDeps.java]: Checking for existence of the same `path` twice? I seems that we already know it does not exists here

File: src/main/java/com/gluonhq/substrate/util/FileDeps.java (hunk 2 above)
Comment (jperedadnr) [reply src/main/java/com/gluonhq/substrate/util/FileDeps.java]: Fixed

File: src/main/java/com/gluonhq/substrate/SubstrateDispatcher.java (hunk 3 above)
Comment (jperedadnr) [reply src/main/java/com/gluonhq/substrate/SubstrateDispatcher.java]: ok

File: src/main/java/com/gluonhq/substrate/target/IosTargetConfiguration.java (hunk 5 above)
Comment (jperedadnr) [reply src/main/java/com/gluonhq/substrate/target/IosTargetConfiguration.java]: ok, refactored

File: src/main/java/com/gluonhq/substrate/target/IosTargetConfiguration.java (hunk 4 above)
Comment (jperedadnr) [reply src/main/java/com/gluonhq/substrate/target/IosTargetConfiguration.java]: ok, refactored

File: src/main/java/com/gluonhq/substrate/util/FileDeps.java (hunk 6 above)
Comment (jperedadnr) [reply src/main/java/com/gluonhq/substrate/util/FileDeps.java]: Right, `setupDependencies` does the same check. Removed it

**Pull request merged**